src/dex/maverick-v2/maverick-v2.ts

```diff
@@ -262,44 +258,63 @@ export class MaverickV2 extends SimpleExchange implements IDex<MaverickV2Data> {
     _: Context,
     executorAddress: Address,
   ): DexExchangeParam {
-    // TODO:
-    if (side === SwapSide.BUY) throw new Error(`Buy not supported`);
-
     const { pool } = data;
 
     srcToken = this.dexHelper.config.wrapETH(srcToken);
     destToken = this.dexHelper.config.wrapETH(destToken);
 
-    const exchangeData = this.maverickV2Iface.encodeFunctionData('swap', [
-      recipient,
-      {
-        amount: side === SwapSide.SELL ? srcAmount : destAmount,
-        tokenAIn: data.tokenA.toLowerCase() === srcToken.toLowerCase(),
-        // exactOutput: side === SwapSide.BUY,
-        exactOutput: false,
-        tickLimit:
-          data.tokenA.toLowerCase() === srcToken.toLowerCase()
-            ? BigInt(data.activeTick) + 100n
-            : BigInt(data.activeTick) - 100n,
-      },
-      '0x',
-    ]);
+    if (side === SwapSide.SELL) {
+      // Perform direct swap for SELL side
+
+      const exchangeData = this.maverickV2Iface.encodeFunctionData('swap', [
+        recipient,
+        {
+          amount: srcAmount,
+          tokenAIn: data.tokenA.toLowerCase() === srcToken.toLowerCase(),
+          exactOutput: false,
+          tickLimit:
+            data.tokenA.toLowerCase() === srcToken.toLowerCase()
+              ? BigInt(data.activeTick) + 100n
+              : BigInt(data.activeTick) - 100n,
+        },
+        '0x',
+      ]);
+
+      return {
+        needWrapNative: this.needWrapNative,
+        transferSrcTokenBeforeSwap: pool,
+        skipApproval: true,
+        targetExchange: pool,
+        dexFuncHasRecipient: true,
+        exchangeData,
+        returnAmountPos: extractReturnAmountPosition(
+          this.maverickV2Iface,
+          'swap',
+          'amountOut',
+        ),
+      };
+    }
+
+    // perform "exactOutputSingleMinimal" via MaverickV2's Router
+    const exchangeData = this.maverickV2RouterIface.encodeFunctionData(
+      'exactOutputSingleMinimal',
+      [
+        recipient,
+        pool,
+        data.tokenA.toLowerCase() === srcToken.toLowerCase(),
+        destAmount,
+        data.tokenA.toLowerCase() === srcToken.toLowerCase()
+          ? BigInt(data.activeTick) + 100n
+          : BigInt(data.activeTick) - 100n,
+      ],
+    );
 
     return {
       needWrapNative: this.needWrapNative,
-      transferSrcTokenBeforeSwap: pool,
-      // skipApproval: true,
-      targetExchange: pool,
+      targetExchange: this.config.routerAddress,
       dexFuncHasRecipient: true,
       exchangeData,
-      returnAmountPos:
-        side === SwapSide.SELL
-          ? extractReturnAmountPosition(
-              this.maverickV2Iface,
-              'swap',
-              'amountOut',
-            )
-          : undefined,
+      returnAmountPos: undefined,
```

Comment:
I think `returnAmountPos` is supported on Ex3 now

Reply:
added `returnAmountPos` from router